frontend/tests/features/wheresMyGene/tissueAutoExpand.test.ts

```diff
@@ -0,0 +1,212 @@
+    await filterCellType(page);
+    await checkTissues(page);
+    await removeCellFilter(page);
+    await checkTissues(page, ["lung"], []);
+    await checkCellTypes(page, cells);
+  });
+});
+/**
+ * *******************************************
+ * Helper Functions
+ * *******************************************
+ */
+
+/**
+ * loadPageAndTissues
+ * Load the WMG page and wait for the tissue nodes to be visible
+ */
+async function loadPageAndTissues(page: Page) {
+  await goToWMG(page);
+  await expect(page.getByTestId(TISSUE_NODE_TEST_ID)).not.toHaveCount(0);
+}
+
+/**
+ * clickIntoFilter
+ * Click into the filter and wait for the tooltip to be visible
+ */
+async function clickIntoFilter(page: Page, filterName: string) {
+  await page
+    .getByTestId(filterName)
+    .getByRole("button", { name: TISSUE_FILTER_LABEL, exact: true })
+    .click();
+  await page.getByRole("tooltip").waitFor();
+}
+
+/**
+ * filterCellType
+ * Filter cell type 'B cell' and exit filter mode
+ */
+async function filterCellType(page: Page, count = 3) {
+  await page
+    .getByTestId(CELL_TYPE_FILTER_TEST_ID)
+    .getByRole("combobox")
+    .click();
+  for (let i = 0; i < count; i++) {
+    await page
+      .getByRole("option", { name: CELL_TYPE_FILTERS[i], exact: true })
+      .click();
+  }
+  await page.keyboard.press("Escape");
+}
+
+/**
+ * filterTissues
+ * Filter the tissuesfrom the left panel
+ */
+async function filterTissues(
+  page: Page,
+  filteredTissues: string[] = FILTERED_TISSUES
+) {
+  await clickIntoFilter(page, TISSUE_FILTER_TEST_ID);
+  for (const tissue of filteredTissues) {
+    await page.getByRole("option", { name: tissue, exact: true }).click();
+  }
+  await page.keyboard.press("Escape");
+}
+
+/**
+ * checkTissues
+ * Check that only filtered tissues are visible, and expanded tissues are expanded
+ */
+async function checkTissues(
+  page: Page,
+  filteredTissues: string[] = FILTERED_TISSUES,
+  expandedTissues = filteredTissues
+) {
+  if (filteredTissues.length !== 0) {
+    await expect(page.getByTestId("tissue-name")).toHaveCount(
+      filteredTissues.length
+    );
+  }
+
+  let i = 0;
+  for (const tissue of expandedTissues) {
+    await expect(page.getByTestId(`cell-type-labels-${tissue}`)).toBeVisible();
+    const height = await page
+      .getByTestId(`cell-type-labels-${tissue}`)
+      .getAttribute("height");
+    toInteger(height) > 20 && i++;
+  }
+  await expect(i).toEqual(expandedTissues.length);
+}
+
+/**
+ * checkCellTypes
+ * Check that all cells under expanded tissues are 'B Cell'
+ */
+async function checkCellTypes(page: Page, cellTypes = CELL_TYPE_FILTERS) {
+  const cells = await page.getByTestId("cell-type-name").allInnerTexts();
+  for (const cell of cells) {
+    await expect(cell).toEqual(cellTypes[indexOf(cells, cell)]);
+  }
+}
+
+/**
+ * removeCellFilter
+ * Remove cell filter
+ */
+async function removeCellFilter(page: Page) {
+  await page.getByTestId("CancelIcon").nth(0).click();
```

Comment:
this scope is a little large, which risks clicking on the cancelIcon in another component in the future, maybe let's scope it to `data-testid="cell-type-search"`?

And I think this case it's okay to use first(), since we don't care what cell type label we delete!

```ts
await page.getByTestId("cell-type-search").getByTestId("CancelIcon").first.click(); 
```

Reply:
thanks @tihuan I ended up doing this which seems to work well:

`async function removeCellFilter(page: Page) {
  await await page
    .getByTestId(`cell-type-tag-${CELL_TYPE_FILTERS[0]}`)
    .getByTestId("CancelIcon")
    .click();
}`